src/gui/uithememanager.cpp

```diff
@@ -140,3 +208,36 @@ QString UIThemeManager::getIconPath(const QString &iconId) const
 #endif
     return IconProvider::instance()->getIconPath(iconId);
 }
+
+void UIThemeManager::loadCustomColorsFromJSONConfig(const QString &configFileName)
+{
+    QFile configFile(configFileName);
+    if (!configFile.open(QIODevice::ReadOnly)) {
+        LogMsg(tr("Failed to open \"%1\", Reason: %2").arg(configFileName, configFile.errorString()), Log::WARNING);
+        return;
+    }
+
+    QJsonParseError jsonError;
+    const QJsonDocument configJsonDoc = QJsonDocument::fromJson(configFile.readAll(), &jsonError);
+    if (jsonError.error != QJsonParseError::NoError) {
+        LogMsg(tr("Invalid JSON provided in \"%1\", Reason: %2").arg(configFileName, jsonError.errorString()), Log::WARNING);
+        return;
+    }
+    if (!configJsonDoc.isObject())
+        LogMsg(tr("Invalid JSON provided in \"%1\", Reason: %2").arg(configFileName, tr("Root JSON value is not an object")), Log::WARNING);
+
+    const QJsonObject colors = configJsonDoc.object().value("colors").toObject();
+    if (colors.isEmpty()) {
+        LogMsg(tr("Can't find any colors in provided JSON config file, QBittorrent will use default colors"), Log::WARNING);
```

Comment:
**q**Bittorrent

Reply:
IMO, it's correct case if no colors are provided in theme file. It can be conceived this way. I would drop this check at all.